ergo-wallet/src/test/scala/org/ergoplatform/wallet/interpreter/ErgoProvingInterpreterSpec.scala

```diff
@@ -35,4 +40,23 @@ class ErgoProvingInterpreterSpec
         }
     }
   }
+
+  it should "sign 50 simple inputs with default cost limit" in {
+    val prover = ErgoProvingInterpreter(obtainSecretKey(), parameters)
+    val pk = prover.hdPubKeys.head
+
+    val value = 100000000L
+    val creationHeight = 10000
+    val boxCandidate = new ErgoBoxCandidate(value, pk, creationHeight)
+
+    val numOfInputs = 50
+    val fakeTxId = ModifierId @@ Base16.encode(Array.fill(32)(5: Byte))
+    val inputBoxes = (1 to numOfInputs).map(i => boxCandidate.toBox(fakeTxId, i.toShort))
+    val unsignedInputs = inputBoxes.map(ib => new UnsignedInput(ib.id, ContextExtension.empty))
+
+    val utx = new UnsignedErgoLikeTransaction(unsignedInputs, IndexedSeq.empty, IndexedSeq(boxCandidate))
+    val signRes = prover.sign(utx, inputBoxes, IndexedSeq(), stateContext)
+    signRes.isSuccess shouldBe true
+  }
```

Comment:
With previous EPI implementation, test failed because of improper costing. Even more, ErgoAppKit-based urlwallet fails on less inputs. 

50 is also about max number of inputs of the current box collector (val maxInputs = 48)